dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/relation/RelationModel.java

```diff
@@ -0,0 +1,93 @@
+package com.dat3m.dartagnan.verification.model.relation;
+
+import com.dat3m.dartagnan.wmm.Relation;
+import com.dat3m.dartagnan.verification.model.event.EventModel;
+
+import java.util.*;
+
+
+public class RelationModel {
+    private final Relation relation;
+    private final List<String> names;
```

Comment:
Why do we need a name? While the relation object can have several alias names, for the model (the name will probably be only used for displaying) I think we only care about one name

Reply:
Why do we need to store any names at all? The relation already provides its name (if any), and if you want to display a different name, then this should be responsibility of the visualizer not the model, no?
I don't think this class (and related one's) should have any code related to displaying/visualizing unless the whole `ExecutionModel(Next)` is meant to be only used for that purpose.

Reply:
Storing the name is necessary because in the case of relation redefinition like following:
```
let ext = ext & ((~IW) * M)
```
The relation with the name `ext` and the one with `ext#2` both exist in `Wmm`. And I think if the user asks to show "ext", we should extract the one with `ext#2`, right?

Reply:
> There will be two different Relations in Wmm: one with the name ext and one with ext#2 and r8. If the user asks to show "ext" and "r8", we should extract the Relation with ext#2 and r8, right? So we have to store the name ext and r8 for the RelationModel whose Relation is ext#2 actually

I would disagree. There is only a single relation, namely `ext#2`/`r8`. If the user requests to visualize `r8`, then he will get `ext#2` displayed but with `r8` as labels. If he requests `ext`, he should get `ext#2` displayed (the most recent version). If he requests both, he either gets only `ext#2` displayed but with two labels, or he gets `ext#2` two times with different labels (that's up to you).
But notice that these are two different things: displaying an existing relation but with a different name (`ext#2` vs. 'r8') vs. resolving the name to another relation (`ext#2` vs. `ext`).

>One more thing: the visualizer does not have information about which relation has to be shown in the graph. 

Well, then this is the canonical place to fix it. The model should not workaround issues of code that uses it (e.g. visualizer, or even CAAT) but the users should adapt to whatever the model is.
After all, the model exists and makes sense without any of those "users", but the users cannot exist without the model. The dependency is thus from model towards users and the code should reflect this.

Lastly, if you want to annotate relations that shall be displayed, then this should happen on `Relation` and not `RelationModel`. We discussed this shortly here #742. See my post about using something like `Map<Relation, ShowAsInfo>` in `Wmm` to associate to a relation display information (which can include different names, colors, arrow types etc.).

Reply:
>I would disagree. There is only a single relation, namely ext#2/r8. If the user requests to visualize r8, then he will get ext#2 displayed but with r8 as labels. If he requests ext, he should get ext#2 displayed (the most recent version). If he requests both, he either gets only ext#2 displayed but with two labels, or he gets ext#2 two times with different labels (that's up to you).

I agree with you in general. But the truth for the case I mentioned is that `ext` and `ext#2` co-exist. I know that the original one should be removed, but it isn't. I discussed it with @hernanponcedeleon and found there are indeed different cases. For example, `rfe`, defined in `stdlib.cat`, included into `vmm.cat` and redefined there. So in the constructed `Wmm` there is only `rfe#2` and NO `rfe`. But for `ext` it is different. Actually if you add a new definition of it using `ext` again (like the definition in the code block above), you can find there are even three versions of it in `Wmm`: `ext`, `ext#2` and `ext#3`.

Reply:
> I agree with you in general. But the truth for the case I mentioned is that ext and ext#2 co-exist. I know that the original one should be removed, but it isn't

Of course the original also exists and it is even used, namely in the definition of `ext#2` (i.e., you cannot remove it). What I meant by "there exists only a single relation" is with regard to `r8` and `ext#2`: those two are different names for the same relation (though whether they are one or two relations shouldn't matter to the user).
The original relation `ext` is not accessible to the user anymore and he will simply have no way to display it via command line options (he could via `show ext as ...` from within the cat-file though).
In particular, if the user requests `ext` to be displayed via commandline, this `ext` should resolve to the most recent version (*) of `ext`, meaning `ext#<maxId>`. To make this easier, it might be reasonable to attach some metadata to the relations during parsing time (e.g., "Original Name" and "Version Number"). In some sense, this would be similar to source location information we maintain for events.
For the time being, it would also be fine to extract original name/version number from the relation names (i.e., assuming the names are of the shape `originalName` and `originalName#versionNumber`)

(*) I think it would even fine to let the user pick the version to be displayed: `ext` would resolve to the most recent version, and `ext#N` would refer to the N-th definition (including `N=0` for the original one).